File: 14-longest-common-prefix/14-longest-common-prefix.go
import "strings"

func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	prefix := strs[0]

	for i := 1; i < len(strs); i++ {
		for strings.Index(strs[i], prefix) != 0 {
			prefix = string([]rune(prefix)[:len(prefix)-1])
		}
	}

	return prefix
}

File: 14-longest-common-prefix/14-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        if not strs:
            return ""
        
        smallestStr = strs[0]
        
        for s in strs:
            if len(s) < len(smallestStr):
                smallestStr = s
            
        i = 0
        
        while i < len(smallestStr):
            curr = smallestStr[i]
            
            for st in strs:
                if curr != st[i]:
                    return smallestStr[:i]
                
            i += 1
            
        return smallestStr[:i]
                
            
        